src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.cs

```diff
@@ -0,0 +1,1195 @@
+
+		private const int c_noValueSetSentinel = -1;
+
+		private static UISettings _uiSettings = null;
+
+		private DispatcherHelper m_dispatcherHelper;
+
+		// Private members
+		TextBlock m_captionTextBlock;
+
+		CompositionPropertySet m_sharedPointerPropertySet;
+
+		private StackPanel m_backgroundStackPanel;
+		private StackPanel m_foregroundStackPanel;
+
+		private bool m_isPointerOver = false;
+		private bool m_isPointerDown = false;
+		private double m_mousePercentage = 0.0;
+
+		private RatingInfoType m_infoType = RatingInfoType.Font;
+		private double m_preEngagementValue = 0.0;
+		private bool m_disengagedWithA = false;
+		private bool m_shouldDiscardValue = true;
+		private long m_fontFamilyChangedToken;
+
+		public RatingControl()
+		{
+			m_dispatcherHelper = new DispatcherHelper(this);
+			//__RP_Marker_ClassById(RuntimeProfiler.ProfId_RatingControl);
+
+			DefaultStyleKey = typeof(RatingControl);
+		}
+
+		~RatingControl()
+		{
+			// We only need to use safe_get in the deruction loop
+			RecycleEvents(true /* useSafeGet */);
+		}
+
+		public event TypedEventHandler<RatingControl, object> ValueChanged;
+
+		private float RenderingRatingFontSize()
+		{
+			// MSFT #10030063 Replacing with Rating size DPs
+			return (float)(c_defaultRatingFontSizeForRendering * GetUISettings().TextScaleFactor);
+		}
+
+		private float ActualRatingFontSize()
+		{
+			return RenderingRatingFontSize() / 2;
+		}
+
+		// TODO MSFT #10030063: Convert to itemspacing DP
+		private double ItemSpacing()
+		{
+			// Stars are rendered 2x size and we use expression animation to shrink them down to desired size,
+			// which will create those spacings (not system margin).
+			// Since text scale factor won't affect system margins,
+			// when stars get bigger, the spacing will become smaller.
+			// Therefore we should include TextScaleFactor when calculating item spacing
+			// in order to get correct total width and star center positions.
+			double defaultFontSize = c_defaultRatingFontSizeForRendering / 2;
+			return c_defaultItemSpacing - (GetUISettings().TextScaleFactor - 1.0) * defaultFontSize / 2;
+		}
+
+		private void UpdateCaptionMargins()
+		{
+			// We manually set margins to caption text to make it center-aligned with the stars
+			// because star vertical center is 0.8 instead of the normal 0.5.
+			// When text scale changes we need to update top margin to make the text follow start center.
+			var captionTextBlock = m_captionTextBlock;
+			if (captionTextBlock != null)
+			{
+				double textScaleFactor = GetUISettings().TextScaleFactor;
+				Thickness margin = captionTextBlock.Margin;
+				margin.Top = c_defaultCaptionTopMargin - (ActualRatingFontSize() * c_verticalScaleAnimationCenterPoint);
+
+				captionTextBlock.Margin(margin);
+			}
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			RecycleEvents();
+
+			// Retrieve pointers to stable controls 
+			//IControlProtected thisAsControlProtected = this;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)